config/supermicro.go

```diff
@@ -114,13 +125,208 @@ func (cm *supermicroVendorConfig) Marshal() (string, error) {
 	}
 }
 
+func (cm *supermicroVendorConfig) Unmarshal(cfgData string) (err error) {
```

Comment:
can you include test cases and some fixture data for these

Reply:
I'd like to produce a future PR with test cases for all the config stuff and not block this one for now.